buffer.go

```diff
@@ -326,22 +332,43 @@ func (b *buffer) resize(size int) {
 
 func (b *buffer) clone() (clone *buffer) {
```

Comment:
Unrelated to the content of this PR, I was curious where this was used, but it's not used anywhere so should we delete it @achille-roussel ?